src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts

```diff
@@ -0,0 +1,388 @@
+        SELF.fetchSubmissionCounts(this.challenge['id'], phase['id']);
+      }
+    };
+  }
+
+  /**
+   * Fetch submissions from API.
+   * @param challenge  challenge id
+   * @param phase  phase id
+   */
+  fetchSubmissions(challenge, phase) {
+    const API_PATH = this.endpointsService.allChallengeSubmissionURL(challenge, phase);
+    const SELF = this;
+    this.apiService.getUrl(API_PATH).subscribe(
+      data => {
+        SELF.submissions = data['results'];
+        SELF.paginationDetails.next = data.next;
+        SELF.paginationDetails.previous = data.previous;
+        SELF.paginationDetails.totalPage = Math.ceil(data.count / 100);
+
+        if (data.count === 0) {
+          SELF.paginationDetails.showPagination = false;
+          SELF.paginationDetails.paginationMessage = 'No results found';
+        } else {
+          SELF.paginationDetails.showPagination = true;
+          SELF.paginationDetails.paginationMessage = '';
+        }
+
+        // condition for pagination
+        if (data.next === null) {
+          SELF.paginationDetails.isNext = 'disabled';
+          SELF.paginationDetails.currentPage = 1;
+        } else {
+          SELF.paginationDetails.isNext = '';
+          SELF.paginationDetails.currentPage = Math.ceil(data.next.split('page=')[1] - 1);
+        }
+        if (data.previous === null) {
+          SELF.paginationDetails.isPrev = 'disabled';
+        } else {
+          SELF.paginationDetails.isPrev = '';
+        }
+      },
+      err => {
+        SELF.globalService.handleApiError(err);
+      },
+      () => {
+        console.log('Fetched submissions', challenge, phase);
+      }
+    );
+  }
+
+  /**
+   * Download Submission csv.
+   */
+  downloadSubmission() {
+    if (this.challenge['id'] && this.selectedPhase && this.fileSelected) {
+      const API_PATH = this.endpointsService.challengeSubmissionDownloadURL(
+        this.challenge['id'], this.selectedPhase['id'], this.fileSelected
+      );
+      const SELF = this;
+      if (SELF.fieldsToGetExport.length === 0 || SELF.fieldsToGetExport === undefined) {
+        SELF.apiService.getUrl(API_PATH, false).subscribe(
+          data => {
+            SELF.windowService.downloadFile(data, 'all_submissions.csv');
+          },
+          err => {
+            SELF.globalService.handleApiError(err);
+          },
+          () => {
+            console.log('Download complete.', SELF.challenge['id'], SELF.selectedPhase['id']);
+          }
+        );
+      }
+    } else {
+      if (this.selectedPhase === null) {
+        this.globalService.showToast('error', 'Please select a challenge phase!');
+      } else if (this.fileSelected === '') {
+        this.globalService.showToast('error', 'The file type requested is not valid!');
+      }
+    }
+  }
+
+  /**
+   * load data with pagination
+   */
+  loadPaginationData = function(url) {
+    if (url !== null) {
+      const SELF = this;
+      const API_PATH = url.split('localhost:8000/api/')[1];
```

Comment:
Get this `localhost:8000/api/` from environment file.

Reply:
@lunayach  done